### PR TITLE
Ignore empty balances

### LINKED ISSUE #2738: Premium stats: Unused location loopring in value distribution by location
## Task

Fix it

### PR DESCRIPTION
**Pull request by yabirgb**
Closes #2738

### CODE REVIEW

File: docs/changelog.rst [hunk 1]
```diff
@@ -38,6 +38,7 @@ Changelog
 * :bug:`2656` Users of coinbase with a lot of assets or trades should now see all of them again. There should be no missing balances or trades thanks to a fix at query pagination.
 * :bug:`2690` Eth2 stakers that have very recently deposited and don't have a validator index yet will now be handled properly and their balance should be shown.
 * :bug:`2716` Users will now get a correct exported CSV file when a sell is matched with multiple acquisitions.
+* :bug:`2738` Premium users won't see empty balances in the statistics for value distribution by location.
```
Comment (LefterisJP) [inline docs/changelog.rst:41]: It's not about balances but locations.
```suggestion
* :bug:`2738` Premium users won't see locations that have no balances in the statistics for value distribution by location.
```

File: rotkehlchen/rotkehlchen.py [hunk 2]
```diff
@@ -806,7 +807,9 @@ def query_balances(
 
         # retrieve loopring balances if module is activated
         if self.chain_manager.get_module('loopring'):
-            balances[str(Location.LOOPRING)] = self.chain_manager.get_loopring_balances()
+            loopring_balances = self.chain_manager.get_loopring_balances()
+            if len(loopring_balances):
```
Comment (LefterisJP) [inline rotkehlchen/rotkehlchen.py:811]: explicit. Never compare 0 to a Falsy value. This is a source of many many bugs.

```suggestion
            if len(loopring_balances) != 0:
```

File: rotkehlchen/rotkehlchen.py [hunk 3]
```diff
@@ -797,7 +797,8 @@ def query_balances(
                 force_token_detection=ignore_cache,
                 ignore_cache=ignore_cache,
             )
-            balances[str(Location.BLOCKCHAIN)] = blockchain_result.totals.assets
+            if len(blockchain_result.totals.assets):
```
Comment (LefterisJP) [inline rotkehlchen/rotkehlchen.py:800]: explicit. Never compare 0 to a Falsy value. This is a source of many many bugs.
```suggestion
            if len(blockchain_result.totals.assets) != 0:
```

File: rotkehlchen/rotkehlchen.py [hunk 4]
```diff
@@ -797,7 +797,8 @@ def query_balances(
...
+                balances[str(Location.BLOCKCHAIN)] = blockchain_result.totals.assets
```
Comment (LefterisJP) [inline rotkehlchen/rotkehlchen.py:801]: We need to ask @kelsos here. Kelso you don't read the all balance query, right? So in the case of no blockchain balances the blockchain key missing is fine in that endpoint, right?

**Review by LefterisJP — COMMENTED** [review]
You misunderstood the problem. This is about the premium statistics and specifically the value distribution by location. It is not calculated  from `query_balances()`

Indeed the problem is first introduced by the balances saving a 0 value even if nothing exists so I think your change is not bad.

But it does not solve the problem. The function that returns latest value by distribution is: `get_latest_location_value_distribution()` and this is what shows the data in the statistics.

And the change you need there is the following, since the data is already saved in the DB. 

```diff
diff --git a/rotkehlchen/db/dbhandler.py b/rotkehlchen/db/dbhandler.py
index 988591f8d..43e16169e 100644
--- a/rotkehlchen/db/dbhandler.py
+++ b/rotkehlchen/db/dbhandler.py
@@ -3133,7 +3133,7 @@ class DBHandler:
         cursor = self.conn.cursor()
         results = cursor.execute(
             'SELECT time, location, usd_value FROM timed_location_data WHERE '
-            'time=(SELECT MAX(time) FROM timed_location_data);',
+            'time=(SELECT MAX(time) FROM timed_location_data) AND usd_value!=0;',
         )
         results = results.fetchall()
```

This is something you can already test since you have a premium key. Make sure to test your approaches to something. If you had done so you would have seen what you did was not enough to solve the issue.

File: rotkehlchen/rotkehlchen.py (hunk 4 above)
Comment (kelsos) [reply rotkehlchen/rotkehlchen.py:801]: All balance is the one that creates the snapshot right? if yes, then that is correct the frontend ignores the result of this query.

**Comment by yabirgb** [comment]
@LefterisJP I tried this one before pushing and it did what I expected, that's why I created the PR :sweat_smile: My idea was that is better to don't add the data rather than modify that query. Indeed changing the query fixes the issue but addressing the root seemed better for me (and indeed it worked).

**Comment by LefterisJP** [comment]
> @LefterisJP I tried this one before pushing and it did what I expected, that's why I created the PR  My idea was that is better to don't add the data rather than modify that query. Indeed changing the query fixes the issue but addressing the root seemed better for me (and indeed it worked).

Would not have worked if you had snapshotted balances as my example has.

File: rotkehlchen/rotkehlchen.py (hunk 4 above)
Comment (LefterisJP) [reply rotkehlchen/rotkehlchen.py:801]: thank you kelsos

**Review by LefterisJP — APPROVED** [review]
Thank you @yabirgb. LGTM now!

**Pull request merged**